stdlib/public/Concurrency/TaskGroup.swift

```diff
@@ -335,31 +329,7 @@ public struct TaskGroup<ChildTaskResult: Sendable> {
   ///         collected += value
   ///     }
   ///
```

Comment:
we lost the thread-safety section which is worrying? We must explain that this should not be called from other tasks, correctness relies on it.